core/src/main/java/org/elasticsearch/index/query/QueryParseContext.java

```diff
@@ -173,33 +172,29 @@ public QueryBuilder parseInnerQueryBuilder() throws IOException {
      */
     @Nullable
     public QueryBuilder parseInnerFilterToQueryBuilder() throws IOException {
-        final boolean originalIsFilter = isFilter;
+        final boolean originalIsFilter = this.shardContext.isFilter;
         try {
-            isFilter = true;
+            this.shardContext.isFilter = true;
             return parseInnerQueryBuilder();
         } finally {
-            isFilter = originalIsFilter;
+            this.shardContext.isFilter = originalIsFilter;
         }
     }
 
     QueryBuilder parseInnerFilterToQueryBuilder(String queryName) throws IOException, QueryParsingException {
```

Comment:
I tried removing them already, the problem is that even though we might have already refactored a query (like AndQueryBuilder), we still need to use `parseInnerFilterToQueryBuilder()` for now because the inner queries might not have been refactored yet and still do their check of the filter flag in parse(). We might be able to replace `parseInnerFilterToQueryBuilder()` with `parseInnterQueryBuilder` after FuzzyQuery and TermsQuery are refactored, but I'm not sure.